src/main/java/com/attacktimer/AttackTimerMetronomePlugin.java

```diff
@@ -25,8 +71,8 @@ public class AttackTimerMetronomePlugin extends Plugin
 {
     public enum AttackState {
         NOT_ATTACKING,
+        DELAYED_FIRST_TICK,
```

Comment:
Is this state when you are flinched?

Reply:
This is not the state when you are flinched as you figured out. 

This is the state we set for the first tick of any attack. This is the only state in which the attack speed can be update more than once, through the `onVarClientIntChanged` subscriber. As if the state is not `DELAYED_FIRST_TICK` then `onVarClientIntChanged` cannot change the speed of the weapon. 

But if the state is `DELAYED_FIRST_TICK`  then the subscriber will re-call `performAttack()` hence updating the speed of the weapon as the player swaps.